### ISSUE TITLE
Feature request: Blur Dithering

### ISSUE THREAD

**Issue by SebiTimeWaster** [photopea/photopea#1860]
Since PhotoPea only works in 8 bit mode you get color bands when you blur a picture (and probably with color gradients and other desctructive filters also?).

It would be nice to have either an 16-bit edit mode (I acknowledge that might not be feasible due to memory restrictions and it would also need an aditional dithering on rendering it on an 8-bit monitor) or to implement a proper 8-bit color dithering when using these features.

example for color banding:
![pea](https://user-images.githubusercontent.com/3726309/71145987-c4d95a80-2223-11ea-81f1-790f72179b9a.png)

**Comment by photopea** [comment]
Hi, yes, I am aware of this problem. Dithering is a possible solution. I will try to investigate it a bit further.

**Comment by photopea** [comment]
Hi, so I drew the same gradient in Photoshop and Photopea, and increased the contrast artifficially. Here are the results. Are you sure dithering is the right approach? It feels wrong to get non-grayscale pixels when drawing grayscale gradients. Noise also has bad impact on the compression.
![image](https://user-images.githubusercontent.com/15955558/71377355-d96f7700-25c4-11ea-9957-6fd18cd4c86b.gif)

**Comment by SebiTimeWaster** [comment]
as to the colored dithering, i have no clue why PS does this, i presume they found it to show a better picture quality? also this looks like some kind of random dithering, which i did not expect...

as to if it is the right approach, i think it is the only approach, i don't think there is anything else one can do to hide or mask this banding effect...

about compression: i rather have a bigger picture than steps in my color gradients.

and further thinking about it, the optimal solution would be if the dithering would only be applied to:
* the picture shown on the monitor
* the "general" or "composited" (no clue how it is called) picture that is saved in the PSD document (not on the layers)
* picture exports (to PNG and such).
this would preserve all details in the PSD while showing it correctly on 8-bit monitors.

but for now, Happy Holidays! and you should not work but take some time off :smiley:

**Comment by photopea** [comment]
Hi, I implemented dithering for gradients. It is about 4x slower than gradients without dithering, but it is still very fast. 

If we did dithering "without randomness", and all rows of the image are the same, all rows would be the same even after dithering (as each row is identical, it would be processed in the same way).

Below is a radial gradient with increased contrast in Photopea and Photoshop. I think Photopea works better.

![pea](https://user-images.githubusercontent.com/15955558/71756160-7ddb9e00-2e8e-11ea-9de5-6513a133830d.gif)
